stan/math/rev/fun/singular_values.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef MATH_SINGULAR_VALUES_H
+#define MATH_SINGULAR_VALUES_H
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+#include <stan/math/rev/core/arena_matrix.hpp>
+#include <stan/math/rev/fun/value_of.hpp>
+#include <stan/math/rev/fun/typedefs.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/err/check_symmetric.hpp>
+#include <stan/math/prim/err/check_nonzero_size.hpp>
+#include <stan/math/prim/fun/typedefs.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+#include <Eigen/SVD>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the singular values of the specified matrix.
+ *
+ * Equation (4) from Differentiable Programming Tensor Networks
+ * is used for MxN input matrix's adjoint calculation.
+ *
+ * @tparam EigMat type of input matrix
+ * @param m MxN input matrix
+ * @return Singular values of matrix
+ */
+template <typename EigMat,
+    require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+	  require_vt_var<EigMat>* = nullptr>
+inline auto singular_values(const EigMat& m) {
+  using ret_type = promote_scalar_t<var, Eigen::VectorXd>;
+  check_nonzero_size("singular_values", "m", m);
+
+  const int M = to_arena(m.rows());
+  auto arena_m = to_arena(m);
+  
+  Eigen::MatrixXd m_val = value_of(arena_m);
+  Eigen::JacobiSVD<Eigen::MatrixXd> svd(m_val, Eigen::ComputeFullU | Eigen::ComputeFullV);
+
+  auto singular_values = to_arena(svd.singularValues());
```

Comment:
`svd.singularValues` will be an `Eigen::VectorXd`

To make this be an autodiff variable, we need to say:

`ret_type singular_values = to_arena(svd.singularValues());`